### ISSUE TITLE
Question: See list of shared inboxes a user has access to

### ISSUE THREAD

**Issue by Digi-Deli** [IBM/aspera-cli#77]
Hi there,

is there a way to get a list of shared inboxes that a user has access to? I tried `ascli aoc packages shared_inboxes list` and unless I set my profile to have a default workspace this command throws an error. The help is not that helpful as it show very basic information about aoc command but not about packages.

When I set my profile with a default workspace
<img width="631" alt="Screenshot 2022-06-08 at 09 25 18" src="https://user-images.githubusercontent.com/103125566/172569812-ca04f90b-9895-4613-a408-4b180c5046fc.png">
When I set my profile without a default workspace
<img width="447" alt="Screenshot 2022-06-08 at 09 25 54" src="https://user-images.githubusercontent.com/103125566/172569930-b10e1d67-daea-4b84-876d-6b16b7c55ea6.png">

It would be nice to see a list of shared inboxes if a user has access to more than one.

Regards,
Digital-Delivery@Arm.com

**Comment by laurent-martin** [comment]
Hi, do you mean listing all shared inboxes of all workspaces you belong to ?

with a unix shell you can do:

```bash
ascli aoc user workspac list --fields=name --format=csv|while read w;do ascli aoc --workspace="$w" packages shared_inboxes list; done
```

**Comment by Digi-Deli** [comment]
Hi Laurent,

thanks for your response. The shell script kind of works although it prompts me to input the PEM pass phrase for each of the workspaces I have access to. Is there a preset in the config file to add the PEM pass phrase so the commands don't prompt the user to input the pass? I have seen the password option but that seems to be for the server command

I won't consider the other two options as I want our users to use the cli instead of doing direct API calls, not many are technical.

**Comment by laurent-martin** [comment]
Well, you are right, private key passphrase option is only supported in the faspex5 plugin.
It will be in aoc plugin in 4.8.0

Here the latest dev version: <https://ibm.biz/ascli_dev>

**Comment by Digi-Deli** [comment]
Hi Laurent,

and what about the fact that I need to set a default workspace to be able to see results when running any of packages commands

for instance when running 
`ascli aoc packages show OBU1opxkAg`

if I have a default workspace (doesn't need to be the workspace where package OBU1opxkAg was uploaded) this command shows me the package information but when I set the default workspace to none the same command does not show any and it throws a 404 error

**Comment by laurent-martin** [comment]
Yes, right, somme commands may work without specifying a workspace, like listing packages or shared inboxes.
Next version will allow specifying `--workspace=@json:null`
beta available at same link as above

**Comment by Digi-Deli** [comment]
It seems to me that these commands make an extra call to get the default workspace. I don't think it's very useful to see the default workspace when you are requesting information for a package that has been uploaded to a different workspace (using packages command). In fact I was confused when I started using the new aspera cli with the output showing me a workspace that had nothing to do with a package I was trying to get the information from

Running show command on a package with default workspace set

`ascli aoc packages show OBU1opxkAg`
![ascli_aoc_package_show](https://user-images.githubusercontent.com/103125566/173071438-3de3b865-9b9b-40d4-b765-f8219205fd0d.png)

`ascli aoc user workspaces list`
![Screenshot 2022-06-10 at 14 09 52](https://user-images.githubusercontent.com/103125566/173071629-40535249-924e-44f6-86eb-bf4b2592e80f.png)

Running show command on a package without default workspace set
`ascli aoc user profile show`
![Screenshot 2022-06-10 at 14 14 40](https://user-images.githubusercontent.com/103125566/173072224-e2b50db4-6571-4de4-979b-e8b3ffdfa5df.png)

`ascli aoc package show OBU1opxkAg`
![Screenshot 2022-06-10 at 14 12 53](https://user-images.githubusercontent.com/103125566/173072324-a8ab42c5-76ae-4714-acc1-6a88d48b3eeb.png)

Same happens when trying to download the package (`ascli aoc packages recv OBU1opxkAg --to-folder=.`), list the packages of a workspace (`ascli aoc packages list --query=@json:'{"workspace_id": 58970}'`), ...

Another observation I found using packages command

When running `ascli aoc packages list --query=@json:'{"workspace_id": "58970"}'`  (with a default workspace set otherwise it throws 404)
![Screenshot 2022-06-10 at 14 52 06](https://user-images.githubusercontent.com/103125566/173080525-86949484-bd98-47b3-ad50-2cc1a9a04c9e.png)

When running `ascli aoc packages list --workspace="Delivery for Mac Environment Company(0002) UK"`
![Screenshot 2022-06-10 at 14 53 13](https://user-images.githubusercontent.com/103125566/173080756-e2862833-c833-414f-936a-c47f2d662ddf.png)

I know all this isn't related to the original question of seen all shared inboxes a user has access to but more to the usage of the packages command

**Comment by laurent-martin** [comment]
Well,

`ascli` tries to remain close to the web ui, where there is a separation of workspaces and the "packages" and "files" applications live inside a workspace.

The fact of using `--query=@json:'{"workspace_id": 58970}'` is left to advanced users, as it modifies the default query that is build for the current workspace which is normally selected by specifying --workspace=xxx , of if not defined: use thwe default.
And yes, if there is no default, it was failing ... (fixed) anyway it is best to have a default workspace or to define one in config or command line.
